elements/lisk-chain/src/block.ts

```diff
@@ -13,49 +13,61 @@
  */
 
 import { codec } from '@liskhq/lisk-codec';
+import { BlockAssets } from './block_assets';
 import { BlockHeader } from './block_header';
 import { blockSchema } from './schema';
 import { Transaction } from './transaction';
 
 interface BlockAttrs {
 	header: Buffer;
 	payload: Buffer[];
+	assets: Buffer[];
 }
 
 export class Block {
 	// eslint-disable-next-line no-useless-constructor
-	public constructor(public readonly header: BlockHeader, public readonly payload: Transaction[]) {
+	public constructor(
+		public readonly header: BlockHeader,
+		public readonly payload: Transaction[],
+		public readonly assets: BlockAssets,
```

Comment:
A bit unsure here if it can be redefined to be something like `BlockAsset[]`. In the LIP it says `assets` is an array of objects and also since payload is an array of objects too.

Reply:
other properties are also class instance, and we could create `BlockAsset` class, but we want to have `blockAssets.get` and `blockAssets.set`. Therefore it's better to be `BlockAssets` class